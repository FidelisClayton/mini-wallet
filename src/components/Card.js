import React from 'react'
import { css } from 'emotion'

import * as colors from '../helpers/colors'

const styles = css({
  width: '80%',
  backgroundColor: colors.white,
  paddingBottom: '25px',
  paddingLeft: '15px',
  paddingRight: '15px',
  boxShadow: colors.cardShadow,
  zIndex: 1,

  '.card': {
    '&__title': {
      color: colors.darkGrey,
      paddingTop: '15px',
      marginTop: '0',
      fontSize: '1.2rem',

      '&--bottom': {
        marginBottom: 0,
        paddingBottom: '15px'
      }
    },

    '&__submit': {
      marginTop: '30px',
    }
  },
})

const Card = ({
  className,
  children,
  ...props
}) => {
  return (
    <div
      className={`${styles} ${className} card`}
      {...props}
    >
      { children }
    </div>
  )
}

export default Card
