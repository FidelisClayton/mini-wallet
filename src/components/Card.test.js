import React from 'react'
import { shallow } from 'enzyme'

import Card from './Card'

describe('Component: <Card />', () => {
  it ('should render properly', () => {
    const wrapper = shallow(
      <Card>
        Hello
      </Card>
    )

    expect(wrapper).toMatchSnapshot()
  })
})
