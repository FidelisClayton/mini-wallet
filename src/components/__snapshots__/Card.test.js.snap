// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: <Card /> should render properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card>
    Hello
</Card>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Hello",
      "className": "css-1du1xnj undefined card",
    },
    "ref": null,
    "rendered": "Hello",
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Hello",
        "className": "css-1du1xnj undefined card",
      },
      "ref": null,
      "rendered": "Hello",
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
